runhouse/resources/hardware/cluster.py

```diff
@@ -1957,3 +1964,195 @@ def _folder_mv(
+                if isinstance(last_active_at, str)
+                else datetime.datetime(1970, 1, 1)
+            )  # Convert to datetime
+            last_active_at = last_active_at.replace(tzinfo=datetime.timezone.utc)
+
+            if cluster_status == "running" and not last_active_at:
+                # For BC, in case there are clusters that were saved and created before we introduced sending cluster status to den.
+                cluster_status = "unknown"
+
+            cluster_info = {
+                "Name": cluster_name,
+                "Cluster Type": cluster_type,
+                "Status": cluster_status,
+                "Last Active (UTC)": last_active_at,
+            }
+            running_clusters.append(
+                cluster_info
+            ) if cluster_status == "running" else not_running_clusters.append(
+                cluster_info
+            )
+
+        # TODO: will be used if we'll need to print not-running clusters
+
+        # Sort clusters by the 'Last Active (UTC)' and 'Status' column
+        not_running_clusters = sorted(
+            not_running_clusters,
+            key=lambda x: (x["Last Active (UTC)"], x["Status"]),
+            reverse=True,
+        )
+        running_clusters = sorted(
+            running_clusters, key=lambda x: x["Last Active (UTC)"], reverse=True
+        )
+
+        return running_clusters, not_running_clusters
+
+    ###############################
+    # Cluster list
+    ###############################
+    @staticmethod
+    def list(
+        get_all_clusters: Optional[bool] = False,
+        since: Optional[bool] = None,
+        status: Optional[Union[str, ResourceServerStatus]] = None,
+    ):
+
+        import sky
+
+        cluster_filters = (
+            Cluster._parse_filters(since=since, cluster_status=status)
+            if not get_all_clusters
+            else {"all": "all"}
+        )
+
+        # get clusters from den
+        den_clusters_resp = Cluster.get_clusters_from_den(
+            cluster_filters=cluster_filters
+        )
+        if den_clusters_resp.status_code != 200:
+            logger.error(f"Failed to load {rns_client.username}'s clusters from Den")
+            den_clusters = []
+        else:
+            den_clusters = den_clusters_resp.json().get("data")
+
+        # get sky live clusters
+        sky_live_clusters = sky.status(refresh=True)
+
+        if not sky_live_clusters and not den_clusters:
+            return [], [], []
+
+        sky_live_clusters = Cluster._get_unsaved_live_clusters(
+            den_clusters=den_clusters, sky_live_clusters=sky_live_clusters
+        )
+
+        (
+            running_clusters,
+            not_running_clusters,
+        ) = Cluster._get_running_and_not_running_clusters(clusters=den_clusters)
+        all_clusters = running_clusters + not_running_clusters
+
+        return all_clusters, running_clusters, sky_live_clusters
```

Comment:
wondering if it makes more sense to return a single dict instead of seaparate lists, wdyt?

Reply:
Don't really have a strong opinion about that, we could return a single dict, probably it will be more elegant to do so.